server/core/include/irods/irods_notify_service_manager.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef IRODS_NOTIFY_SERVICE_MANAGER_HPP
+#define IRODS_NOTIFY_SERVICE_MANAGER_HPP
+
+#include <fmt/format.h>
+#include <fmt/compile.h>
+
+#include <cstddef>
+#include <ctime>
+#include <string>
+#include <utility>
+
+namespace irods
+{
+    void notify_service_manager(const std::size_t _msg_length, const char* _msg, const char* _socket_path);
```

Comment:
Please make `_msg` the first parameter and `_msg_length` the second parameter. The order of those parameter surprised me. I suspect readers of the source may find that uncommon as well.

Reply:
This is done intentionally to disambiguate it from the wrappers defined below.

Reply:
Ah, so these function are not part of the public interface.

In that case, consider wrapping the private components in a detail namespace or renaming them in such a way that it's clear which components make up the public interface. For example:
- Use a detail namespace
- Or, add a prefix to private functions (e.g. `do_`)
- Or, add a suffix to private functions (e.g. `_impl`)

That'll allow you to change the order of the parameters.

Reply:
Done